tests/model_fields/test_jsonfield.py

```diff
@@ -1006,8 +1006,13 @@ def test_key_sql_injection(self):
                 ).exists(),
                 False,
             )
+        cast = (
+            "::jsonb"
+            if connection.vendor == "postgresql" and connection.is_psycopg3
```

Comment:
Please try to minimize `connection.vendor` usage. At least this one is unfriendly to the CockroachDB backend.

Reply:
ACK, in this case I can probably simply remove the cast from the test. Do you have any recommendations from the past on how to solve those? `hasattr` check on `is_psycopg3`?

Reply:
Encapsulate any `connection.vendor` logic in a `DatabaseFeatures` attribute.